src/sqlancer/transformations/Transformation.java

```diff
@@ -0,0 +1,116 @@
+package sqlancer.transformations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+import net.sf.jsqlparser.statement.Statement;
+
+/**
+ * The base class of transformations Defines APIs to remove, replace, remove elements of a list.
+ */
+public class Transformation {
+
+    private static Supplier<Boolean> bugJudgement;
+    protected boolean isChanged;
+    String current;
+    Statement statement;
+    // public enum StatusCode {
```

Comment:
Do we need this commented-out code?

Reply:
> Do we need this commented-out code?

Currently not.
`name` is originally designed to identify the Transformation. (But seems not necessary at present).
`StatusCode` is designed to identify the status (if it succeeded or failed) after applying transformation. But it seems that `boolean` can already handle it.